release/llm_tests/serve/test_llm_serve_correctness.py

```diff
@@ -0,0 +1,210 @@
+import os
+import subprocess
+import time
+import requests
+import pytest
+from openai import OpenAI
+from ray import serve
+from ray.serve.llm import LLMConfig, build_openai_app, ModelLoadingConfig
+
+MODEL_ID = "Qwen/Qwen2.5-0.5B-Instruct"
+RAY_MODEL_ID = "qwen-0.5b"
+MAX_OUTPUT_TOKENS = 256
+SEED = 42
+
+# vLLM has started using this with V1 instead of xgrammar
+os.environ["RAYLLM_GUIDED_DECODING_BACKEND"] = "auto"
+
+
+def get_llm_config(tensor_parallel_size=1, pipeline_parallel_size=1):
+    """Create LLMConfig with specified parallelism parameters."""
+    return LLMConfig(
+        model_loading_config=ModelLoadingConfig(
+            model_id=RAY_MODEL_ID,
+            model_source=MODEL_ID,
+        ),
+        deployment_config=dict(
+            autoscaling_config=dict(
+                min_replicas=1,
+                max_replicas=1,
+            )
+        ),
+        engine_kwargs=dict(
+            tensor_parallel_size=tensor_parallel_size,
+            pipeline_parallel_size=pipeline_parallel_size,
+        ),
+        runtime_env=None,
+    )
+
+
+def start_ray_serve(tensor_parallel_size=1, pipeline_parallel_size=1):
+    """Start Ray Serve with specified parallelism parameters."""
+    ray_url = "http://localhost:8000"
+    llm_config = get_llm_config(tensor_parallel_size, pipeline_parallel_size)
+    app = build_openai_app({"llm_configs": [llm_config]})
+    serve.run(app, blocking=False)
+    return ray_url
+
+
+def start_vllm_server(tensor_parallel_size=1, pipeline_parallel_size=1):
+    """Start vLLM server with specified parallelism parameters."""
+    vllm_port = 8001
+    cmd = [
+        "vllm",
+        "serve",
+        MODEL_ID,
+        "--port",
+        str(vllm_port),
+        "--distributed-executor-backend=ray",
+        "--generation-config=vllm",  # Force vLLM to ignore HF generation_config.json
+        "--tensor-parallel-size",
+        str(tensor_parallel_size),
+        "--pipeline-parallel-size",
+        str(pipeline_parallel_size),
+    ]
+    process = subprocess.Popen(cmd)
+    return f"http://localhost:{vllm_port}", process
+
+
+def generate_with_ray(test_prompt, ray_serve_llm_url):
+    openai_api_base = f"{ray_serve_llm_url}/v1"
+    client = OpenAI(base_url=openai_api_base, api_key="fake-key")
+
+    response = client.completions.create(
```

Comment:
I would use both chat and completion to make sure prompt formatting is applied properly too.